app/lib/features/onboarding/pages/redeem_invitations_page.dart

```diff
@@ -0,0 +1,355 @@
+        ),
+      ],
+    );
+  }
+
+  Widget _buildNavigationButtons(BuildContext context, L10n lang) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        showGetDetailsButton
+            ? _buildGetDetailsButton(context, lang)
+            : const SizedBox.shrink(),
+        const SizedBox(height: 16),
+        showRedeemButton
+            ? Column(
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: [
+                _buildRedeemButton(context, lang),
+                const SizedBox(height: 16),
+              ],
+            )
+            : const SizedBox.shrink(),
+        if (!showRedeemButton) _buildSkipButton(context, lang),
+        const SizedBox(height: 16),
+      ],
+    );
+  }
+
+  Widget _buildGetDetailsButton(BuildContext context, L10n lang) {
+    return ElevatedButton(
+      onPressed:
+          _tokenController.text.isNotEmpty
+              ? () {
+                if (!_formKey.currentState!.validate()) return;
+                if (!inCI && !ref.read(hasNetworkProvider)) {
+                  showNoInternetNotification(context);
+                  return;
+                }
+                setState(() {
+                  showInviteInfo = true;
+                });
+              }
+              : null,
+      child: Text(lang.getDetails, style: const TextStyle(fontSize: 16)),
+    );
+  }
+
+  Widget _buildRedeemButton(BuildContext context, L10n lang) {
+    return ActerPrimaryActionButton(
+      onPressed:
+          _tokenController.text.isNotEmpty
+              ? () {
+                if (!_formKey.currentState!.validate()) return;
+                if (!inCI && !ref.read(hasNetworkProvider)) {
+                  showNoInternetNotification(context);
+                  return;
+                }
+                redeemToken(context, ref, _tokenController.text.trim());
+              }
+              : null,
+      child: Text(lang.redeem, style: const TextStyle(fontSize: 16)),
+    );
+  }
+
+  Widget _buildSkipButton(BuildContext context, L10n lang) {
+    return OutlinedButton(
+      onPressed: () {
+        EasyLoading.dismiss();
+        context.goNamed(
+          Routes.saveUsername.name,
+          queryParameters: {'username': widget.username},
+        );
+      },
+      child: Text(lang.skip),
+    );
+  }
+
+  void redeemToken(BuildContext context, WidgetRef ref, String token) async {
+    final lang = L10n.of(context);
+    final superInvites = await ref.read(superInvitesProvider.future);
+
+    EasyLoading.show(status: lang.redeeming(token));
+    try {
+      final rooms = (await superInvites.redeem(token)).toList();
+      if (!context.mounted) {
+        EasyLoading.dismiss();
+        return;
+      }
+      EasyLoading.showToast(lang.addedToSpacesAndChats(rooms.length));
+      context.goNamed(
+          Routes.saveUsername.name,
+          queryParameters: {'username': widget.username},
+        );
```

Comment:
It shouldn't redirect to next screen upon redemption.  User can redeem multiple code if he/she wants,

```suggestion
      context.goNamed(
          Routes.saveUsername.name,
          queryParameters: {'username': widget.username},
        );
```

Reply:
Got it.